spec/unit/pdk/util/puppet_version_spec.rb

```diff
@@ -24,9 +24,22 @@
   shared_context 'is a package install' do
     before(:each) do
       allow(PDK::Util).to receive(:package_install?).and_return(true)
+      allow(PDK::Util::RubyVersion).to receive(:versions).and_return('2.1.9' => '2.1.0', '2.4.3' => '2.4.0')
 
-      mock_response = cache_versions.map { |r| Gem::Version.new(r) }
-      allow(PDK::Util::RubyVersion).to receive(:available_puppet_versions).and_return(mock_response)
+      PDK::Util::RubyVersion.use('2.1.9')
+      instance219 = PDK::Util::RubyVersion.instance
+      PDK::Util::RubyVersion.use('2.4.3')
+      instance243 = PDK::Util::RubyVersion.instance
```

Comment:
This feels like it could benefit from a `.instance(version)`, instead of requiring two calls?

Reply:
:+1: